streams/src/main/java/org/apache/kafka/streams/state/internals/ChangeLoggingKeyValueBytesStore.java

```diff
@@ -145,4 +145,9 @@ public void putAll(final List<KeyValue<Bytes, byte[]>> entries) {
     void log(final Bytes key, final byte[] value) {
         context.logChange(name(), key, value, context.timestamp(), position);
     }
+
+    @Override
+    public Position getPosition() {
+        return position;
```

Comment:
Similar concern here... It seems like we have to merge this layer's position with the wrapped store's position to compute a correct result for this method.

Reply:
The changelogging stores don't have a position anymore since I added the public `getPosition`. The merging now is necessary only in the caching layer